src/main/kotlin/org/jitsi/nlj/rtcp/KeyframeRequester.kt

```diff
@@ -51,32 +51,44 @@ class KeyframeRequester : TransformerNode("Keyframe Requester") {
     var waitIntervalMs = DEFAULT_WAIT_INTERVAL_MS
 
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
-        val packet = packetInfo.packet
-        val pliOrFir = when (packet) {
-            is CompoundRtcpPacket -> {
-                packet.packets.first { it is RtcpFbPliPacket || it is RtcpFbFirPacket } as? RtcpFbPacket
+        val packet = packetInfo.packet.apply {
```

Comment:
Why do this as apply and then a when on this instead of a when on the packet?

Reply:
Because with `when` I don't get an "it" or "this" reference (I keep running into this, it's annoying), which means I have to access the field as `packetInfo.packet`. But I can't do that because "packetInfo.packet is a mutable property that could have changedby this time". Or am I missing something?

Reply:
On mobile so little hard to tell, but why can't it stay how it was?

Reply:
Just to avoid having the extra `val`

Reply:
This is worse than an extra val, in my opinion.